src/v/v8_engine/data_policy.h

```diff
@@ -40,6 +40,8 @@ class data_policy {
 public:
     static constexpr int8_t version{1};
 
+    data_policy() = default; // Need for boost::lexical_cast
```

Comment:
because of `std::istream& operator>>(std::istream& i, data_policy& datapolicy);` ? it might be cleaner to avoid this, or use some sort of factory method so we don't need a default constructor.

Reply:
```
namespace boost 
{
    template <typename Target, typename Source>
    inline Target lexical_cast(const Source &arg)
    {
        Target result = Target();

        if (!boost::conversion::detail::try_lexical_convert(arg, result)) {
            boost::conversion::detail::throw_bad_cast<Source, Target>();
        }

        return result;
    }
```

Boost call default constructor inside

Reply:
seems like you could remove the `std::istream` overload and invoke json parsing directly.